### PR TITLE
boards: Add initial support for the nucleo-f446re board.

### PR DESCRIPTION
**Pull request by aakkds**
Tested hello_world sample

Signed-off-by: Philémon Jaermann <p.jaermann@gmail.com>

### CODE REVIEW

File: boards/arm/nucleo_f446re/doc/nucleof446re.rst [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+********
+
+The Nucleo F446RE board features an ARM Cortex-M4 based STM32F446RE MCU
+with a wide range of connectivity support and configurations. Here are
+some highlights of the Nucleo F446RE board:
+
+
+- STM32 microcontroller in QFP64 package
+- Two types of extension resources:
+       - Arduino Uno V3 connectivity
```
Comment (dbkinder) [inline boards/arm/nucleo_f446re/doc/nucleof446re.rst]: need a blank line before a (sub) list like this.  Also don't indent it so far or the doc processor treats this as a block quote.  The ``-`` should be aligned with the text of the outdented list above it:

    - Two types of extension resources:

      - Arduino Uno V3 Connectivity
      - ST morpho ...

File: boards/arm/nucleo_f446re/doc/nucleof446re.rst [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+       - ST morpho extension pin headers for full access to all STM32 I/Os
+- On-board ST-LINK/V2-1 debugger/programmer with SWD connector
+- Flexible board power supply:
+       - USB VBUS or external source(3.3V, 5V, 7 - 12V)
```
Comment (dbkinder) [inline boards/arm/nucleo_f446re/doc/nucleof446re.rst]: same here, blank line before, reduce indent

File: boards/arm/nucleo_f446re/doc/nucleof446re.rst [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+- Quad SPI(1)
+- Camera Interface
+- GPIO(50) with external interrupt capability
+- 12-bit ADC(3) with 16 channels
+- 12-bit DAC with 2 channels
+
+
+
+More information about STM32F446RE can be found here:
+       - `STM32F446RE on www.st.com`_
```
Comment (dbkinder) [inline boards/arm/nucleo_f446re/doc/nucleof446re.rst]: blank line before, reduce indent

File: boards/arm/nucleo_f446re/doc/nucleof446re.rst [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+- I2C1_SCL : PB8
+- I2C2_SDA : PB3
+- I2C2_SCL : PB10
+- I2C3_SDA : PB4
+- I2C3_SCL : PA8
+
+System Clock
+------------
+
+Nucleo F446RE System Clock could be driven by internal or external oscillator,
```
Comment (dbkinder) [inline boards/arm/nucleo_f446re/doc/nucleof446re.rst]: by an internal

File: boards/arm/nucleo_f446re/doc/nucleof446re.rst [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+as well as main PLL clock. By default System clock is driven by PLL clock at 84MHz,
```
Comment (dbkinder) [inline boards/arm/nucleo_f446re/doc/nucleof446re.rst]: Some more missing articles:

    as the main

    By default, the System clock is driven by the PLL clock at 84MHz, driven by an
    8MHz high-speed external clock.

File: boards/arm/nucleo_f446re/doc/nucleof446re.rst [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+
+Applications for the ``nucleo_f446re`` board configuration can be built and
+flashed in the usual way (see :ref:`build_an_application` and
+:ref:`application_run` for more details).
+
+Flashing
+========
+
+Nucleo F446RE board includes an ST-LINK/V2-1 embedded debug tool interface.
+This interface is supported by the openocd version included in Zephyr SDK.
```
Comment (dbkinder) [inline boards/arm/nucleo_f446re/doc/nucleof446re.rst]: in the Zephyr SDK.

**Review by dbkinder — CHANGES_REQUESTED** [review]
a few wording and formatting fixes

**Comment by aakkds** [comment]
@dbkinder Thanks for the review! I updated the PR to take your comments into account.

File: arch/arm/soc/st_stm32/stm32f4/Kconfig.soc [hunk 7]
```diff
@@ -39,6 +39,9 @@ config SOC_STM32F429XI
 config SOC_STM32F469XI
 	bool "STM32F469XI"
 
+config SOC_STM32F446XE
```
Comment (erwango) [inline arch/arm/soc/st_stm32/stm32f4/Kconfig.soc]: Use numerical order

File: boards/arm/nucleo_f446re/Kconfig.defconfig [hunk 8]
```diff
@@ -0,0 +1,32 @@
+# Kconfig - STM32F446RE Nucleo board configuration
+#
+# Copyright (c) 2018 Philémon Jaermann
+#
+# Taken from nucleo_f411re:
```
Comment (erwango) [inline boards/arm/nucleo_f446re/Kconfig.defconfig]: This is nice but remove these. Apply everywhere

File: dts/arm/st/mem.h [hunk 9]
```diff
@@ -88,6 +88,9 @@
 #elif defined(CONFIG_SOC_STM32L432XC)
 #define DT_FLASH_SIZE		__SIZE_K(256)
 #define DT_SRAM_SIZE		__SIZE_K(64)
+#elif defined(CONFIG_SOC_STM32F446XE)
```
Comment (erwango) [inline dts/arm/st/mem.h]: Use alphanumerical order

File: dts/arm/st/stm32f446.dtsi [hunk 10]
```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright (c) 2018 Philémon Jaermann
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <st/stm32f4.dtsi>
```
Comment (erwango) [inline dts/arm/st/stm32f446.dtsi]: #include <st/stm32f401.dtsi>

**Review by erwango — CHANGES_REQUESTED** [review]
Some comments and need to split into 2 commits:
-SoC
-Board
Otherwise looks good

**Comment by aakkds** [comment]
@erwango Thanks for your input! I updated the PR.

**Review by galak — COMMENTED** [review]
Looks good to me, will merge when @erwango and @dbkinder approve it.

**Review by dbkinder — DISMISSED** [review]
Thanks.

**Review by erwango — DISMISSED** [review]

File: arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.stm32f446xe [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+if SOC_STM32F446XE
+
+config SOC
+	string
+	default stm32f446xx
+
+config NUM_IRQS
+	int
+	default 96
+
```
Comment (ydamigos) [inline arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.stm32f446xe]: I would add here:
```
if GPIO_STM32

config GPIO_STM32_PORTE
	default n

config GPIO_STM32_PORTH
	default n

endif # GPIO_STM32
```

**Review by ydamigos — COMMENTED** [review]
Just a non-blocking comment from my side.

File: arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.stm32f446xe (hunk 11 above)
Comment (galak) [reply arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.stm32f446xe]: I made this change, and pushed an update

**Review by galak — APPROVED** [review]

**Pull request merged**